crates/uv/src/commands/pip_show.rs

```diff
@@ -0,0 +1,139 @@
+                "{}{} Please provide a package name or names.",
+                "warning".yellow().bold(),
+                ":".bold(),
+            );
+        }
+        return Ok(ExitStatus::Failure);
+    }
+
+    // Detect the current Python interpreter.
+    let platform = Platform::current()?;
+    let venv = if let Some(python) = python {
+        PythonEnvironment::from_requested_python(python, &platform, cache)?
+    } else if system {
+        PythonEnvironment::from_default_python(&platform, cache)?
+    } else {
+        match PythonEnvironment::from_virtualenv(platform.clone(), cache) {
+            Ok(venv) => venv,
+            Err(uv_interpreter::Error::VenvNotFound) => {
+                PythonEnvironment::from_default_python(&platform, cache)?
+            }
+            Err(err) => return Err(err.into()),
+        }
+    };
+
+    debug!(
+        "Using Python {} environment at {}",
+        venv.interpreter().python_version(),
+        venv.python_executable().simplified_display().cyan()
+    );
+
+    // Build the installed index.
+    let site_packages = SitePackages::from_executable(&venv)?;
+
+    // Sort and deduplicate the packages, which are keyed by name.
+    packages.sort_unstable();
+    packages.dedup();
+
+    // Map to the local distributions.
+    let distributions = {
+        let mut distributions = Vec::with_capacity(packages.len());
+
+        // Identify all packages that are installed.
+        for package in &packages {
+            let installed = site_packages.get_packages(package);
+            if installed.is_empty() {
+                writeln!(
+                    printer,
+                    "{}{} Package(s) not found for: {}",
+                    "warning".yellow().bold(),
+                    ":".bold(),
+                    package.as_ref().bold()
+                )?;
+            } else {
+                distributions.extend(installed);
+            }
+        }
+
+        distributions
+    };
+
+    // Like `pip`, if no packages were found, return a failure.
+    if distributions.is_empty() {
+        return Ok(ExitStatus::Failure);
+    }
+
+    // Print the information for each package.
+    let mut first = true;
+    for distribution in &distributions {
+        if first {
+            first = false;
+        } else {
+            // Print a separator between packages.
+            #[allow(clippy::print_stdout)]
+            {
+                println!("---");
+            }
+        }
+
+        // Print the name, version, and location (e.g., the `site-packages` directory).
+        #[allow(clippy::print_stdout)]
+        {
+            println!("Name: {}", distribution.name());
+            println!("Version: {}", distribution.version());
+            println!(
+                "Location: {}",
+                distribution
+                    .path()
+                    .parent()
+                    .expect("package path is not root")
+                    .simplified_display()
```

Comment:
@ChannyClaus - I think `pip show` _always_ shows the `site-packages` location regardless of whether it's editable. It looks like `pip show` has a separate `Editable location` field for the editable source, so we can add that in a separate PR?

Reply:
Ah, you're absolutely right, thanks for catching that